### PR TITLE
Billing

### PR DESCRIPTION
**Pull request by NureBieloshapkaMykyta**

### CODE REVIEW

File: api/ChattO/API/Controllers/AccountController.cs [hunk 1]
```diff
@@ -30,37 +28,37 @@ public async Task<IActionResult> Login([FromBody] LoginRequest loginRequest)
...
-    {
-        var properties = new AuthenticationProperties
-        {
-            RedirectUri = Url.Action(nameof(GoogleCallBack)),
-            Items =
-            {
-                { "scheme", GoogleDefaults.AuthenticationScheme }
-            }
-        };
+    //[HttpGet("GoogleLogin")]
```
Comment (KatrinaSidorenko) [inline api/ChattO/API/Controllers/AccountController.cs]: Why are these methods commented?

File: api/ChattO/API/Controllers/BillingController.cs [hunk 2]
```diff
@@ -1,27 +1,43 @@
...
     [ProducesResponseType<bool>(StatusCodes.Status200OK)]
     [ProducesResponseType<string>(StatusCodes.Status400BadRequest)]
     [ProducesResponseType(StatusCodes.Status500InternalServerError)]
     [ProducesResponseType(StatusCodes.Status404NotFound)]
-    public async Task<IActionResult> Create([FromBody] CreateBillingRequest request)
+    public async Task<IActionResult> Create([FromQuery] Guid billingInfoId)
     {
-        return Ok();
+        var user = await _userService.GetCurrentUser();
+        var request = new CreateBillingRequest() { OrganizationId = user.Data.OrganizationId, BillingInfoId = billingInfoId };
```
Comment (KatrinaSidorenko) [inline api/ChattO/API/Controllers/BillingController.cs]: Why do we need this additional entity? We can just pass needed properties  to command

File: api/ChattO/Application/BillInfo/Commands/CreateBillingInfo.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        private readonly IRepository<BillingInfo> _repository;
+
+        public Handler(IRepository<BillingInfo> repository)
+        {
+            _repository = repository;
+        }
+
+        public async Task<Result<bool>> Handle(Command request, CancellationToken cancellationToken)
+        {
+            var createResult = await _repository.AddItemAsync(new BillingInfo()
```
Comment (KatrinaSidorenko) [inline api/ChattO/Application/BillInfo/Commands/CreateBillingInfo.cs]: Try to implement an IMapWith<BillingInfo> on the Command class and then just create a map to BillingInfo. Example in CreateOrganization class is provided

File: api/ChattO/Application/BillInfo/Queries/Get.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+{
+    public class Query : IRequest<Result<PagingResponse<BillingInfo>>>
+    {
+        [Filter]
+        public BillingType? Type { get; set; }
+
+        [Filter]
+        public double? Price { get; set; }
+
+        public PagingProps PagingProps { get; set; }
```
Comment (KatrinaSidorenko) [inline api/ChattO/Application/BillInfo/Queries/Get.cs]: We can inherit PagingProps instead.

File: api/ChattO/Application/Payment/Commands/CreateBilling.cs [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        public async Task<Result<CreateOrderResponse>> Handle(Command request, CancellationToken cancellationToken)
+        {
+            var validationResult = await _validator.ValidateAsync(request);
+            if (!validationResult.IsValid)
+            {
+                return Result.Failure<CreateOrderResponse>(validationResult.ToString(" "));
+            }
+
+            var billingId = Guid.NewGuid();
+            var billing = new Billing()
```
Comment (KatrinaSidorenko) [inline api/ChattO/Application/Payment/Commands/CreateBilling.cs]: Here you can also do mapping between Command and Billing

File: api/ChattO/Infrastructure/Extensions/GoogleExtension.cs [hunk 6]
```diff
@@ -11,12 +11,12 @@ public static IServiceCollection AddGoogleAuthentification(this IServiceCollecti
         var clientId = configuration.GetSection("GoogleAuthSettings:ClientId").Get<string>();
         var clientSecret = configuration.GetSection("GoogleAuthSettings:ClientSecret").Get<string>();
 
-        services.AddAuthentication()
-            .AddGoogle(options =>
-            {
-                options.ClientId = clientId;
-                options.ClientSecret = clientSecret;
-            });
+        //services.AddAuthentication();
```
Comment (KatrinaSidorenko) [inline api/ChattO/Infrastructure/Extensions/GoogleExtension.cs]: Why?

File: api/ChattO/Infrastructure/Infrastructure.csproj [hunk 7]
```diff
@@ -19,10 +19,4 @@
     <ProjectReference Include="..\Application\Application.csproj" />
   </ItemGroup>
 
-  <ItemGroup>
```
Comment (KatrinaSidorenko) [inline api/ChattO/Infrastructure/Infrastructure.csproj]: Why is this part removed?

File: api/ChattO/API/Controllers/AccountController.cs (hunk 1 above)
Comment (NureBieloshapkaMykyta) [reply api/ChattO/API/Controllers/AccountController.cs]: fixed

File: api/ChattO/API/Controllers/BillingController.cs (hunk 2 above)
Comment (NureBieloshapkaMykyta) [reply api/ChattO/API/Controllers/BillingController.cs]: done

File: api/ChattO/Application/BillInfo/Commands/CreateBillingInfo.cs (hunk 3 above)
Comment (NureBieloshapkaMykyta) [reply api/ChattO/Application/BillInfo/Commands/CreateBillingInfo.cs]: done

File: api/ChattO/Application/Payment/Commands/CreateBilling.cs (hunk 5 above)
Comment (NureBieloshapkaMykyta) [reply api/ChattO/Application/Payment/Commands/CreateBilling.cs]: done

File: api/ChattO/Infrastructure/Extensions/GoogleExtension.cs (hunk 6 above)
Comment (NureBieloshapkaMykyta) [reply api/ChattO/Infrastructure/Extensions/GoogleExtension.cs]: fixed

File: api/ChattO/Infrastructure/Infrastructure.csproj (hunk 7 above)
Comment (NureBieloshapkaMykyta) [reply api/ChattO/Infrastructure/Infrastructure.csproj]: fixed

**Review by KatrinaSidorenko — APPROVED** [review]
